File: src/data/symbols.js
/**
 * Wheel symbols — what can appear on the roulette.
 *
 * @typedef {object} SymbolDef
 * @property {string} id
 * @property {string} name
 * @property {string} emoji
 * @property {string} color — color group for color bets
 * @property {number} baseValue — innate value when landed on
 * @property {'common'|'uncommon'|'rare'|'legendary'} rarity
 * @property {boolean} startsUnlocked
 * @property {string|null} specialEffect
 */

/** @type {SymbolDef[]} */
export const SYMBOLS = [];

export const SYMBOL_MAP = new Map(SYMBOLS.map(s => [s.id, s]));

export function getSymbol(id) {
  const s = SYMBOL_MAP.get(id);
  if (!s) throw new Error(`Unknown symbol: ${id}`);
  return s;
}

export const RARITY_WEIGHTS = { common: 50, uncommon: 30, rare: 15, legendary: 5 };

File: src/data/symbols.lua
--[[
    Wheel symbols — matches legacy JS symbols.js (empty pool, scored by pocket number).
]]

local SYMBOLS = {}

local SYMBOL_MAP = {}

local function getSymbol(id)
    local s = SYMBOL_MAP[id]
    if not s then error('Unknown symbol: ' .. tostring(id)) end
    return s
end

local RARITY_WEIGHTS = { common = 50, uncommon = 30, rare = 15, legendary = 5 }

return {
    SYMBOLS = SYMBOLS,
    SYMBOL_MAP = SYMBOL_MAP,
    getSymbol = getSymbol,
    RARITY_WEIGHTS = RARITY_WEIGHTS,
}
